File: app/schemas/admin_cuenta.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


PlanCode = Literal["profesional", "consultorio", "centro"]
EstadoSuscripcion = Literal[
    "trial", "active", "past_due", "cancelled", "expired",
]
EstadoCuentaAdmin = Literal[
    "trial", "active", "past_due", "cancelled", "expired",
    "sin_suscripcion",
]


class AdminProfesionalPrincipalRespuesta(BaseModel):
    profesional_id: int
    nombre: str
    apellido: str
    matricula: str
    email: str | None


class AdminCuentaItemRespuesta(BaseModel):
    cuenta_id: int
    nombre: str
    tipo: Literal["individual", "organizacion"]
    plan: PlanCode | None
    estado: EstadoCuentaAdmin
    created_at: datetime
    trial_ends_at: datetime | None
    profesionales_count: int
    miembros_count: int
    profesional_principal: AdminProfesionalPrincipalRespuesta | None


class AdminCuentasPaginaRespuesta(BaseModel):
    items: list[AdminCuentaItemRespuesta]
    total: int
    offset: int
    limit: int


class AdminCuentasResumenRespuesta(BaseModel):
    cuentas_totales: int
    trials_activos: int
    suscripciones_activas: int
    trials_finalizados: int
    altas_ultimos_30_dias: int


class AdminCuentaDatosRespuesta(BaseModel):
    id: int
    nombre: str
    tipo: Literal["individual", "organizacion"]
    created_at: datetime
    updated_at: datetime


class AdminSuscripcionDetalleRespuesta(BaseModel):
    plan: PlanCode
    estado: EstadoSuscripcion
    estado_persistido: EstadoSuscripcion | None
    trial_started_at: datetime | None
    trial_ends_at: datetime | None
    trial_days_remaining: int


class AdminMiembroCuentaRespuesta(BaseModel):
    usuario_id: int
    nombre: str
    email: str
    rol_cuenta: Literal["propietario", "administrador", "miembro"]
    activo: bool
    miembro_desde: datetime


class AdminProfesionalCuentaRespuesta(BaseModel):
    id: int
    nombre: str
    apellido: str
    matricula: str
    email: str | None
    activo: bool


class AdminCuentaDetalleRespuesta(BaseModel):
    cuenta: AdminCuentaDatosRespuesta
    suscripcion: AdminSuscripcionDetalleRespuesta | None
    miembros: list[AdminMiembroCuentaRespuesta]
    profesionales: list[AdminProfesionalCuentaRespuesta]


class AdminAccionMotivoEntrada(BaseModel):
    motivo: str | None = None


class AdminExtenderTrialEntrada(AdminAccionMotivoEntrada):
    dias: Literal[7, 14, 30]


class AdminCambiarPlanEntrada(AdminAccionMotivoEntrada):
    plan: PlanCode


class AdminEventoSuscripcionRespuesta(BaseModel):
    id: int
    actor_usuario_id: int | None
    actor_nombre: str | None
    actor_tipo: Literal["usuario", "sistema"]
    accion: str
    estado_anterior: EstadoSuscripcion | None
    estado_nuevo: EstadoSuscripcion | None
    plan_anterior: PlanCode | None
    plan_nuevo: PlanCode | None
    motivo: str | None
    created_at: datetime

File: frontend/src/types/adminCuenta.ts
export type PlanAdmin = "profesional" | "consultorio" | "centro";
export type EstadoAdmin = "trial" | "active" | "past_due" | "cancelled" | "expired" | "sin_suscripcion";

export interface ProfesionalPrincipalAdmin { profesional_id: number; nombre: string; apellido: string; matricula: string; email: string | null; }
export interface CuentaAdminItem { cuenta_id: number; nombre: string; tipo: "individual" | "organizacion"; plan: PlanAdmin | null; estado: EstadoAdmin; created_at: string; trial_ends_at: string | null; profesionales_count: number; miembros_count: number; profesional_principal: ProfesionalPrincipalAdmin | null; }
export interface CuentasAdminPagina { items: CuentaAdminItem[]; total: number; offset: number; limit: number; }
export interface CuentasAdminResumen { cuentas_totales: number; trials_activos: number; suscripciones_activas: number; trials_finalizados: number; altas_ultimos_30_dias: number; }
export interface CuentaAdminDetalle {
  cuenta: { id: number; nombre: string; tipo: "individual" | "organizacion"; created_at: string; updated_at: string };
  suscripcion: null | { plan: PlanAdmin; estado: Exclude<EstadoAdmin, "sin_suscripcion">; estado_persistido: Exclude<EstadoAdmin, "sin_suscripcion"> | null; trial_started_at: string | null; trial_ends_at: string | null; trial_days_remaining: number };
  miembros: Array<{ usuario_id: number; nombre: string; email: string; rol_cuenta: "propietario" | "administrador" | "miembro"; activo: boolean; miembro_desde: string }>;
  profesionales: Array<{ id: number; nombre: string; apellido: string; matricula: string; email: string | null; activo: boolean }>;
}

export interface FiltrosCuentasAdmin { q?: string; estado?: string; plan?: string; created_from?: string; offset: number; limit: number; }
export interface EventoSuscripcionAdmin { id: number; actor_usuario_id: number | null; actor_nombre: string | null; actor_tipo: "usuario" | "sistema"; accion: string; estado_anterior: Exclude<EstadoAdmin, "sin_suscripcion"> | null; estado_nuevo: Exclude<EstadoAdmin, "sin_suscripcion"> | null; plan_anterior: PlanAdmin | null; plan_nuevo: PlanAdmin | null; motivo: string | null; created_at: string; }
